src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -60,12 +60,19 @@ internal class AnalysisScope
         /// </summary>
         public bool IsPartialAnalysis { get; }
 
+        /// <summary>
+        /// True if we are performing syntactic or semantic analysis for a single source file with a single analyzer in scope,
+        /// which is a <see cref="CompilerDiagnosticAnalyzer"/>.
+        /// </summary>
+        public bool IsSingleFileAnalysisForCompilerAnalyzer =>
```

Comment:
consider naming this `IsSyntacticOrSemanticSingleFileAnalysisForCompilerAnalyzer`

Reply:
(but i'm also ok with this).